datastream-server-restli/src/test/java/com/linkedin/datastream/server/TestCoordinator.java

```diff
@@ -174,6 +183,54 @@ public void teardown() throws IOException {
     _embeddedZookeeper.shutdown();
   }
 
+  @Test
+  public void testRegistersMetricsCorrectly() throws Exception {
+    String testCluster = "testCoordinatorMetrics";
+    Coordinator coordinator = createCoordinator(_zkConnectionString, testCluster);
+    coordinator.start();
+
+    // Make sure the set of metrics the Coordinator registers with the DynamicMetricsManager
+    // matches the metricInfos the Coordinator returns from getMetricInfos().
+    List<BrooklinMetricInfo> metricInfos = coordinator.getMetricInfos();
+    DynamicMetricsManager metricsManager = DynamicMetricsManager.getInstance();
+
+    String coordinatorSimpleClassName = Coordinator.class.getSimpleName();
+    // Keep Coordinator metricInfos
+    Map<String, BrooklinMetricInfo> metricInfoByName = metricInfos.stream()
+        .filter(info -> info.getNameOrRegex().startsWith(coordinatorSimpleClassName))
+        .collect(Collectors.toMap(BrooklinMetricInfo::getNameOrRegex, Function.identity()));
+
+    // Keep Coordinator metrics
+    Map<String, Metric> metrics = metricsManager.getMetricRegistry().getMetrics().entrySet().stream()
+        .filter(entry -> entry.getKey().startsWith(coordinatorSimpleClassName))
+        .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+    Assert.assertEquals(metricInfoByName.size(), metrics.size());
+
+    // Mapping of BrooklinMetricInfo sub-type -> com.codahale.metrics type
+    Map<Class<? extends BrooklinMetricInfo>, Class<? extends Metric>> metricInfoTypeToMetricType = new HashMap<>();
+    metricInfoTypeToMetricType.put(BrooklinCounterInfo.class, Counter.class);
+    metricInfoTypeToMetricType.put(BrooklinGaugeInfo.class, Gauge.class);
+    metricInfoTypeToMetricType.put(BrooklinHistogramInfo.class, Histogram.class);
+    metricInfoTypeToMetricType.put(BrooklinMeterInfo.class, Meter.class);
+
+    for (Map.Entry<String, BrooklinMetricInfo> entry : metricInfoByName.entrySet()) {
+      String metricInfoKey = entry.getKey();
+      BrooklinMetricInfo metricInfo = entry.getValue();
+
+      // For every BrooklinMetricInfo, there must be a registered codahale metric
+      Metric metric = metrics.get(metricInfoKey);
+      Assert.assertNotNull(metric);
+
+      // The registered codahale metric type must correspond to the BrooklinMetricInfo sub-type
+      Class<? extends Metric> actualMetricClazz = metric.getClass();
```

Comment:
any particular reason to name variable actualMetricClazz and not actualMetricClass

Reply:
Java tradition (_sigh_): https://stackoverflow.com/questions/2529974/why-do-java-programmers-like-to-name-a-variable-clazz

Reply:
That code was moved into a dedicated utility for validating metrics. I also changed the code to use `Class` instead of `Clazz` since it's not used on its own but as part of a longer identifier.